Lib/asyncio/base_events.py

```diff
@@ -155,6 +156,63 @@ def _run_until_complete_cb(fut):
     futures._get_loop(fut).stop()
 
 
+
+class _SendfileProtocol(protocols.Protocol):
+    def __init__(self, transp):
+        # transport should be _FlowControlMixin instance
+        self._transport = transp
+        self._proto = transp.get_protocol()
+        self._should_resume_reading = transp.is_reading()
+        self._should_resume_writing = transp._protocol_paused
+        transp.pause_reading()
+        transp.set_protocol(self)
+        if self._should_resume_writing:
+            self._paused = self._transport._loop.create_future()
```

Comment:
Can you add a comment explaining the purpose of `self._paused` future?  I have a hard time fully understanding it.

Reply:
Maybe we can give `self._paused` a better name?

Reply:
Let's rename it to `self._can_write_fut`.

Reply:
`can_write` is about ability to write data and it sounds like a bool flag.
Say, on closing transport *can write* should be logically true.

But the discussed attribute reports about write pausing state.

Reply:
You can call it `_write_ready` or `_write_ready_fut`.

`_paused` is confusing. Paused what? It took me 20 minutes (literally) to figure out how this all works.

Reply:
`_write_ready_fut` sounds good to me.